pkg/controller/worker/state_actuator.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package worker
+
+import (
+	"context"
+	"encoding/json"
+
+	extensionscontroller "github.com/gardener/gardener-extensions/pkg/controller"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	machinev1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/go-logr/logr"
+)
+
+type genericStateActuator struct {
+	logger logr.Logger
+
+	client client.Client
+}
+
+// NewStateActuator creates a new Actuator that reconciles Worker's State subresource
+// It provides a default implementation that allows easier integration of providers.
+func NewStateActuator(logger logr.Logger) StateActuator {
+	return &genericStateActuator{logger: logger.WithName("worker-state-actuator")}
+}
+
+func (a *genericStateActuator) InjectClient(client client.Client) error {
+	a.client = client
+	return nil
+}
+
+// Reconcile update the Worker state with the latest.
+func (a *genericStateActuator) Reconcile(ctx context.Context, worker *extensionsv1alpha1.Worker) error {
+	copyOfWorker := worker.DeepCopy()
+	if err := a.updateWorkerState(ctx, copyOfWorker); err != nil {
+		return errors.Wrapf(err, "failed to update the state in worker status")
+	}
+
+	return nil
+}
+
+func (a *genericStateActuator) updateWorkerState(ctx context.Context, worker *extensionsv1alpha1.Worker) error {
+	state, err := a.getWorkerState(ctx, worker)
+	if err != nil {
+		return err
+	}
+
+	err = extensionscontroller.TryUpdateStatus(ctx, retry.DefaultBackoff, a.client, worker, func() error {
```

Comment:
```suggestion
	return extensionscontroller.TryUpdateStatus(ctx, retry.DefaultBackoff, a.client, worker, func() error {
```

Reply:
I was using the error for some debug logs and forgot the reverse it. Thanks for pointing it.